### PR TITLE
Add a document for build IDs

### PR DESCRIPTION
**Pull request by dschuff**
Describe the `build_id` custom section and its use cases.

### CODE REVIEW

**Comment by dschuff** [comment]
I believe this captures most of our discussion in #133. 
An implementation in lld can be found at https://reviews.llvm.org/D107662

**Review by bkotsopoulossc — APPROVED** [review]
Wooh!

File: BuildId.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+symbolized.
+
+Because the build ID is usually intended to identify debug info, tools that
+transform Wasm binaries will need to decide whether they will preserve, drop, or
+recompute the build ID. Generally speaking, if a transformation would invalidate debug info
+(for example, by rewriting the code section and changing the code offsets), then the tool
+should drop or recompute the build ID. If a tool updates or regenerates debug info along
+with its transformation, it should also generally update the build ID, since the debug
+info no longer matches. Conversely, adding custom sections to a binary would generally
+not require updating the build ID.
```
Comment (sbc100) [inline BuildId.md:18]: These paragraphs looks like they could use a reflow.

File: BuildId.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Build ID
```
Comment (sbc100) [inline BuildId.md:1]: Newline after title lines

File: BuildId.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+## Build ID Section
+The Build ID section is a 
+[custom section](https://webassembly.github.io/spec/core/binary/modules.html#custom-section)
+and thus has no semantic effects and can be stripped at any time. It is named `build_id`
+and has no restriction on where in the binary it can appear. It consists of only 2 fields:
+
+| Field       | Type        | Description |
+| ----------- | ----------- | ----------- |
+| length      | `varuint32` | number of bytes that follow |
+| id          | `bytes`     | sequence of bytes |
```
Comment (sbc100) [inline BuildId.md:29]: Might as well align the final closing column too

**Review by sbc100 — APPROVED** [review]

**Pull request merged**